### ISSUE TITLE
App Installed in 2 Websites Opened in Same browser

### ISSUE THREAD

**Issue by dyodedev** [ohmybrew/laravel-shopify#107]
Hi Team,
I have created a Shopify-Laravel app using your module. I installed the app on 2 different Shopify websites. When I opened the app dashboard of both websites on the same browser, It was quite surprising that the content related to one website was loaded on both the websites.
I think this is a session issue/bug related to the connector.
Can anyone please help me to solve this?
Please help

**Comment by ohmybrew** [comment]
Its supposed to detect the domain difference and do a re-auth sequence. Can you further explain how to duplicate this process.

**Comment by dyodedev** [comment]
My app is used for creating and listing of passcodes. I installed the app on two different Shopify websites. Then I opened the app dashboard in Shopify backend of both the stores in the same browser.
The passcodes which are related to/created for one website appears on both the websites.

**Comment by ptcampbell** [comment]
Is this a local installation? It sounds like both stores are referencing the same database.

**Comment by dyodedev** [comment]
@ptcampbell Yes both stores are referencing the same database. because the app is having only one database.

**Comment by brianakidd** [comment]
@dyodedev Forgive me for stating the obvious, but just checking that when you query your database you are doing so in the context of the shop, correct?

If you have a passcodes table, you'll probably want to have a foreign key for the shop and then you can do something list ShopifyApp::shop()->passcodes to get the passcodes for the current shop.

Is that what you are doing?

**Comment by dyodedev** [comment]
@brianakidd
The problem is not with fetching the values. The problem is with the session. When you open two websites in the same browser, domain of any one store set as session variable in the app.

**Comment by ohmybrew** [comment]
Its the session yes. [Theres code there which checks](https://github.com/ohmybrew/laravel-shopify/blob/master/src/ShopifyApp/Middleware/AuthShop.php#L28) if the store domain does not match whats in session and redirects. I have a test for this which works, so I will have to see whats up with it :/

**Comment by ohmybrew** [comment]
Sorry just gettin back to this, will begin seeing whats up

**Comment by ohmybrew** [comment]
@dyodedev I am not able to reproduce, I have app open in both shops,,

I switch to X shop, refresh, it shows correct data.
I switch back to Y shop, refresh, it shows correct data.

Can you explain what you're exactly doing and how the URLs are setup in the app settings?

**Comment by dyodedev** [comment]
@ohmybrew Let me check again and will revert

**Comment by dyodedev** [comment]
![github_error](https://user-images.githubusercontent.com/40353943/46544688-e762aa80-c8e1-11e8-9244-69585ec6f7f8.png)

We are facing an issue "that trash does not exist".  Can you please he us on this. The screen shot is attached.

**Comment by brianakidd** [comment]
I can't look at it closely right now but trashed is usually associated with model soft deletes.  When are you getting this error?  Navigating to home screen or something else?

**Comment by dyodedev** [comment]
@brianakidd We are facing this issue when loading the home screen.

**Comment by brianakidd** [comment]
Are you using the shop model that comes in the package or did you define your own?  The package is expecting the shop model to have the SoftDeletes trait.

**Comment by ohmybrew** [comment]
Best to extend this packages model if you wish to use your own :) this way
your model will be compatible :)

**Comment by ncpope** [comment]
@dyodedev Here are the instructions for extending the packages Shop model: https://github.com/ohmybrew/laravel-shopify/wiki/Overriding-and-Extending#models 

It sounds like this is related to your issue.

**Comment by tom-it** [comment]
I'm still facing this issue for some reason, I'm running 10.3.1 

After quite some debugging it looks like for some reason the domain is lost in the session.

What happens:
- I login to Store A open my app settings page, all works as expected
- I open store B and open my app settings page
  The reauthorize flow is started
  I see the domain is set in the session, I see the request has the correct shop
 Another redirect follows after this, I see the code returns false from the ShopSession isValid() function, this because $this->getDomain()  == null

I added a workaround, by adding $this->setDomain($this->shop->shopify_domain) in the isValid function. and then I am able to use the app on both my stores without issues.
